crates/bevy_pbr/src/lib.rs

```diff
@@ -67,14 +68,18 @@ pub const SHADOWS_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 11350275143789590502);
 pub const PBR_SHADER_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 4805239651767701046);
+pub const PBR_PREPASS_SHADER_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 9407115064344201137);
 pub const PBR_FUNCTIONS_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 16550102964439850292);
 pub const SHADOW_SHADER_HANDLE: HandleUntyped =
     HandleUntyped::weak_from_u64(Shader::TYPE_UUID, 1836745567947005696);
 
 /// Sets up the entire PBR infrastructure of bevy.
 #[derive(Default)]
-pub struct PbrPlugin;
+pub struct PbrPlugin {
+    pub prepass_enabled: bool,
```

Comment:
This will be off by default: is that desired?

Reply:
Looks like yes, based on a similar struct below.

Reply:
I'm not fully decided on this being off by default or not. Considering a lot of post process effect will need it soon after this is merged we might aswell enable it here. It's just that it has a performance cost on simple scenes and I don't want people to open a basic scene and complain that their fps got lower.

Reply:
Fully post-process effects that use nodes after the main pass could use the depth buffer regardless of whether we do a prepass or not, if they implement their effect via nodes or so. If they use multiple cameras then they are anyway out of luck for getting the depth buffer from a previous camera, I think. However, materials will not have access to depth without a prepass, and that is likely a more common case. However, at the moment I'm leaning toward it being opt-in rather than opt-out, i.e. default disabled.

Reply:
@superdump correct me if I'm wrong, but right now, the depth is not accessible after the main pass because it doesn't have the appropriate flags and no way to customize them.

Considering that a few features like SSAO and TAA will need it and that a lot of people have asked for this feature, I think it makes more sense to make it opt-out at this point. The thing is that you need to configure it on the Material and the camera. So making it true on the Material by default means you only need to activate it on the camera. That way the easy path is to have it turned on and effects like SSAO and TAA will work by default without needing any other manipulation than adding the bundles to the camera. This way disabling the prepass per material becomes more of an optimization for advanced users.